### PR TITLE
add more stats reporting around download retries and failures

### PR DESCRIPTION
**Pull request by craig-day**
Add some retry logic around open and read failures and report that information to statsd client if it is present.

/cc @grosser

### CODE REVIEW

**Comment by craig-day** [comment]
fyi @dadah89

File: lib/i18n/backend/http.rb [hunk 1]
```diff
@@ -123,6 +131,37 @@ def path(locale)
...
+      def record(event, options = {})
+        return unless statsd = @options[:statsd_client]
+
+        case event
+        when :download_fail
+          statsd.increment("#{STATS_NAMESPACE}.download_failed", tags: options[:tags])
+        when :open_retry
+          statsd.increment("#{STATS_NAMESPACE}.open_retry", tags: options[:tags])
+        when :read_retry
+          statsd.increment("#{STATS_NAMESPACE}.read_retry", tags: options[:tags])
```
Comment (grosser) [inline lib/i18n/backend/http.rb]: all very similar ... just send ?

**Review by grosser — APPROVED** [review]

File: lib/i18n/backend/http.rb (hunk 1 above)
Comment (craig-day) [inline lib/i18n/backend/http.rb]: kinda nice to have the raise in the else case, so we can't report random stats. Could check against a set of known things I guess

**Comment by grosser** [comment]
works too ...

**Pull request merged**